### PR TITLE
DATA-573 camera getImage lags

### PR DESCRIPTION
**Pull request by bazile-clyde**
This is a unit test for ReadMedia. That function works as expected despite the comment claiming it's off-by-one. If there's nothing wrong with this function then it's probably the `Next` call from the stream. Will continue to investigate but at the very least this narrows down the surface area of what we'll need to check in the future.

### CODE REVIEW

**Comment by edaniels** [comment]
I'll take a look next week. As an FYI, the ReadMedia comment is saying that you may not get continuous media, not that you won't. It's fairly hard to make it happen.

File: go.mod [hunk 1]
```diff
@@ -26,6 +26,10 @@ require (
 
 require (
 	4d63.com/gochecknoglobals v0.1.0 // indirect
+	cloud.google.com/go v0.102.1 // indirect
```
Comment (edaniels) [inline go.mod:29]: can you run a go mod tidy. not sure why these showed up

File: media_test.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
+package gostream
+
+import (
```
Comment (edaniels) [inline media_test.go:3]: can you run this through make lint

File: media_test.go [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+func (is *ImageSource) Read(_ context.Context) (image.Image, func(), error) {
+	if is.idx >= len(is.Images) {
+		return nil, func() {}, nil
+	}
+	img := is.Images[is.idx]
+	is.idx++
+	return img, func() {}, nil
+}
+
+// Nothing to do.
```
Comment (edaniels) [inline media_test.go:29]: no comment needed

File: media_test.go [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+func (is *ImageSource) Close(_ context.Context) error {
+	return nil
+}
+
+func PNGtoImage(t *testing.T, path string) image.Image {
```
Comment (edaniels) [inline media_test.go:34]: can unexport

File: media_test.go [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+	"context"
+	"github.com/pion/mediadevices/pkg/prop"
+	"go.viam.com/test"
+	"image"
+	"image/png"
+	"os"
+	"testing"
+)
+
+type ImageSource struct {
```
Comment (edaniels) [inline media_test.go:14]: can unexport

File: media_test.go (hunk 4 above)
Comment (edaniels) [reply media_test.go:34]: add a t.Helper here

File: media_test.go [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+	colors := []image.Image{
+		PNGtoImage(t, "data/red.png"),
+		PNGtoImage(t, "data/blue.png"),
+		PNGtoImage(t, "data/green.png"),
+		PNGtoImage(t, "data/yellow.png"),
+		PNGtoImage(t, "data/fuchsia.png"),
+		PNGtoImage(t, "data/cyan.png"),
+	}
+
+	var imgSource ImageSource
```
Comment (edaniels) [inline media_test.go:52]: this isn't a deep copy but shallow. either way, you can do imgSource := imageSource{Images: colors}

File: go.mod (hunk 1 above)
Comment (bazile-clyde) [reply go.mod]: Done

File: media_test.go (hunk 2 above)
Comment (bazile-clyde) [reply media_test.go:3]: Done

File: media_test.go (hunk 5 above)
Comment (bazile-clyde) [reply media_test.go]: Done

File: media_test.go (hunk 3 above)
Comment (bazile-clyde) [reply media_test.go]: Done

File: media_test.go (hunk 4 above)
Comment (bazile-clyde) [reply media_test.go]: Done and Done.

File: media_test.go (hunk 6 above)
Comment (bazile-clyde) [reply media_test.go]: Done. Good catch.

**Pull request merged**